grobid-core/src/main/java/org/grobid/core/jni/JEPThreadPool.java

```diff
@@ -0,0 +1,116 @@
+package org.grobid.core.jni;
+
+import java.util.concurrent.*;  
+import java.util.*;
+import java.io.*;
+
+import org.grobid.core.utilities.GrobidProperties;
+import org.grobid.core.exceptions.GrobidResourceException;
+
+import jep.Jep;
+import jep.JepConfig;
+import jep.JepException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * For using DeLFT deep learning models, we use JEP as JNI CPython interpreter. 
+ * JEP presents the following constraint: A thread that creates a JEP instance 
+ * must be reused for all method calls to that JEP instance. For ensuring this,
+ * we pool the Jep instances in a singleton class. 
+ */
+
+public class JEPThreadPool { 
+    private static final Logger LOGGER = LoggerFactory.getLogger(JEPThreadPool.class);
+
+    private int POOL_SIZE = 1;
+
+    private ExecutorService executor;  
+    private Map<Long,Jep> jepInstances;
```

Comment:
Was there a reason not to use [ThreadLocal](https://docs.oracle.com/javase/8/docs/api/java/lang/ThreadLocal.html)?
I guess originally the intention was to use JEP for multiple threads? (judging by the name `pool`)